Document Dev deployments and its requirements on KIE Sandbox

thiagoelg opened the pull request:
Closes https://github.com/apache/incubator-kie-issues/issues/652

yesamer commented:
Thank you @thiagoelg, that's much better.

Can you please better clarify the image usage with an example?

eg. `It expects a Quarkus application to be running at the same host, which it will use to fetch information to render a form that let's users interact with a Decision.`

That means we need a canvas_dev_deployment_quarkus_blank_app ?

thiagoelg commented:
> Thank you @thiagoelg, that's much better.
> 
> Can you please better clarify the image usage with an example?
> 
> eg. `It expects a Quarkus application to be running at the same host, which it will use to fetch information to render a form that let's users interact with a Decision.`
> 
> That means we need a canvas_dev_deployment_quarkus_blank_app ?

I've fixed some things and added a better README to both the dmn-form-webapp and its image, can you check if it's better now?

thiagoelg commented:
New PR with code changes to `dev-deployment-dmn-form-webapp`: https://github.com/apache/incubator-kie-tools/pull/2438

ljmotta commented on packages/online-editor/docs/DEV_DEPLOYMENTS_ARCHITECTURE.md:
```diff
@@ -0,0 +1,265 @@
+  && microdnf --disableplugin=subscription-manager clean all
+
+# Create the user home, tmp, and .m2 paths to be used by our application.
+# Setting the permission to 777 is important so that our user can access and modify these files during runtime.
```
```suggestion
# Setting the permission to 777 is important so that our users can access and modify these files during runtime.
```

ljmotta commented on packages/online-editor/docs/DEV_DEPLOYMENTS_ARCHITECTURE.md:
```diff
@@ -0,0 +1,265 @@
+1. KIE Sandbox will zip all the files in the workspace being deployed (ignoring .git directories);
+2. Then it will load the chosen deployment option (Custom Image or Kogito Quarkus Blank app) YAML files and modify them dynamically:
+   - Applying the required patches to set KIE Sandbox labels, annotations, and resource names;
+   - Applying patches related to the parameters set in the deployment modal (each parameter correspond to a set of patches and tokens);
```
```suggestion
   - Applying patches related to the parameters set in the deployment modal (each parameter corresponds to a set of patches and tokens);
```

ljmotta commented on packages/online-editor/docs/DEV_DEPLOYMENTS_ARCHITECTURE.md:
```diff
@@ -0,0 +1,265 @@
+
+![Dev Deployment Flow](./images/dev_deployment_flow.png)
+
+The "magic" that drives Dev Deployments depends on various parts working together in sync: KIE Sandbox, your Kubernetes cluster, the container images, de upload service, etc.
```
```suggestion
The "magic" that drives Dev Deployments depends on various parts working together in sync: KIE Sandbox, your Kubernetes cluster, the container images, the upload service, etc.
```

ljmotta reviewed (commented):
@thiagoelg Thanks for the PR. I've got just some minor things to adjust. Great documentation.
